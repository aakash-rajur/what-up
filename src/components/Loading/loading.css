@import "../../utils/variables.css";

@keyframes spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.loading {
    display: inline-block;
    width: 52px;
    height: 52px;
    border-radius: 50%;
    border: 4px solid transparent;
    border-top-color: var(--accent-color);
    border-right-color: var(--accent-color);
    animation: spin 1s linear infinite;
}
