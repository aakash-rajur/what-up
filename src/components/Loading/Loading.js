import React from "react";
import "./loading.css";

function Loading({className}) {
  return <div className={`loading ${className || ""}`}>&nbsp;</div>;
}

export default Loading;
