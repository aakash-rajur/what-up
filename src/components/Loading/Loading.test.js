import React from 'react';
import Loading from './Loading';

describe('Loading component', () => {
    let className = 'additional',
        component = shallow(<Loading className={className}/>);

    it('should render', () => {
        expect(component.find('div.loading').exists())
            .toBe(true);
    });

    it('check props and classNames', () => {
        expect(component.find('div').hasClass('loading')).toBe(true);
        expect(component.find('div').hasClass(className)).toBe(true);
    });

    it('match snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});
