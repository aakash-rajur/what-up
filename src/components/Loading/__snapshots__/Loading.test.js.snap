// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component match snapshot 1`] = `
<div
  className="loading additional"
>
  Â 
</div>
`;
